experimental/framework/data-objects/src/kvpair/DataObject.ts

```diff
@@ -11,44 +11,72 @@ import { IDirectoryValueChanged, IValueChanged } from "@fluidframework/map";
  * IKeyValueDataObject describes the public API surface for our KeyValue DataObject.
  */
 export interface IKeyValueDataObject extends EventEmitter {
-    /**
-     * Get value at Key
-     */
-    get: (key: string) => any;
-
-    /**
-     * Set Value at Key
-     */
-    set: (key: string, value: any) => void;
-
-    /**
-     * Event on value change
-     */
-    on(event: "changed", listener: (args: IDirectoryValueChanged) => void): this;
+  /**
+   * Get value at Key
+   */
+  get: (key: string) => any;
+
+  /**
+   * Set Value at Key
+   */
+  set: (key: string, value: any) => void;
+
+  /**
+   * Event on value change
+   */
+  on(event: "changed", listener: (args: IDirectoryValueChanged) => void): this;
+
+  keys(): string[];
+
+  query(): (test?: string | ((value: string) => boolean)) =>  any | {[key: string]: any}
+
```

Comment:
code style in the repo is four spaces

Reply:
here and elsewhere

Reply:
React is usually 2, but I think we agreed we'd standardize everywhere on 4